plugins/usbdmx/AsyncUsbSender.cpp

```diff
@@ -31,27 +31,29 @@ using ola::usb::LibUsbAdaptor;
 
 AsyncUsbSender::AsyncUsbSender(LibUsbAdaptor *adaptor,
                                libusb_device *usb_device)
-    : AsyncUsbTransceiverBase(adaptor, usb_device),
-      m_pending_tx(false) {
+    : AsyncUsbTransceiverBase(adaptor, usb_device) {
 }
 
 AsyncUsbSender::~AsyncUsbSender() {
   m_adaptor->Close(m_usb_handle);
 }
 
-bool AsyncUsbSender::SendDMX(const DmxBuffer &buffer) {
+bool AsyncUsbSender::SendDMX(const DmxBuffer &buffer, unsigned int portId) {
   if (!m_usb_handle) {
     OLA_WARN << "AsyncUsbSender hasn't been initialized";
     return false;
   }
   ola::thread::MutexLocker locker(&m_mutex);
   if (m_transfer_state == IDLE) {
-    PerformTransfer(buffer);
+    PerformTransfer(buffer, portId);
   } else {
     // Buffer incoming data so we can send it when the outstanding transfers
     // complete.
-    m_pending_tx = true;
-    m_tx_buffer.Set(buffer);
+    if (m_tx_buffers.count(portId)) {
```

Comment:
beware: untested code block. I didn't yet get the USB device to "stall" = to have outstanding transfers at this point.